### ISSUE TITLE
the table doesnt display at the view page

### ISSUE THREAD

**Issue by tamarica** [Chumper/Datatable#289]
i get this one instead of a  table:

<table id="axaYGA1P" class="table table-bordered"> <colgroup> <col class="con0" /> <col class="con1" /> </colgroup> <thead> <tr> <th align="center" valign="middle" class="head1">id</th> <th align="center" valign="middle" class="head2">Name</th> </tr> </thead> <tbody> </tbody> </table> <script type="text/javascript"> jQuery(document).ready(function(){ // dynamic table oTable = jQuery('#axaYGA1P').dataTable({ "sPaginationType": "full_numbers", "bProcessing": false, "sAjaxSource": "http:\/\/localhost:8000\/api\/users", "bServerSide": true, }); // custom values are available via $values array }); </script>

**Comment by Chumper** [comment]
what does your code looks like?

**Comment by tamarica** [comment]
wow its so strange. i wrote the code in this comment and after i clicked on the grin button, it created this small table

**Comment by tamarica** [comment]
<table id="axaYGA1P" class="table table-bordered"> <colgroup> <col class="con0" /> <col class="con1" /> </colgroup> <thead> <tr> <th align="center" valign="middle" class="head1">id</th> <th align="center" valign="middle" class="head2">Name</th> </tr> </thead> <tbody> </tbody> </table> <script type="text/javascript"> jQuery(document).ready(function(){ // dynamic table oTable = jQuery('#axaYGA1P').dataTable({ "sPaginationType": "full_numbers", "bProcessing": false, "sAjaxSource": "http:\/\/localhost:8000\/api\/users", "bServerSide": true, }); // custom values are available via $values array }); </script>

**Comment by tamarica** [comment]
oops/ it did it again/ well, i dont know why i cant see this at the view page and i get only the code

**Comment by Chumper** [comment]
what does your php code in the controller and view look like?

**Comment by tamarica** [comment]
controller looks like this:
thats the function that connect to datatable if i understood corectly
public function getDatatable()
    {
        return Datatable::collection(new Collection(DB::select(DB::raw(
        "SELECT id,first_name  FROM users"))))
            ->showColumns('id', 'first_name')
            ->searchColumns('first_name')
            ->orderColumns('id','first_name')
            ->make();
    }
thats the view page:

{{ Datatable::table()
    ->addColumn('id','Name')
    ->setUrl(route('api.users'))
    ->render() }}

**Comment by Chumper** [comment]
Did you include jquery and datatable.js in your view? 
Do you get any javascript errors?

**Comment by tamarica** [comment]
yes. i include and dont get any errors at javascript. i use by the way Editor. maybe thats makes the problem?

**Comment by Chumper** [comment]
Could be, if you visit the route `api.users` what does the response looks like? 
Should be a json response

**Comment by tamarica** [comment]
i also dont know how to integrate the Editor.but for now i thought it will be enough to display the table only.
the javascript looks like this: 
(function($){

$(document).ready(function() {
	var editor = new $.fn.dataTable.Editor( {
		"ajax": "/api/users",
		"table": "#users",
		"fields": [
			{
				"label": "email",
				"name": "email"
			},
			{
				"label": "first name",
				"name": "first_name"
			},
			{
				"label": "last name",
				"name": "last_name"
			},
			{
				"label": "created at",
				"name": "created_at",
				"type": "date",
				"dateFormat": "D, d M y"
			},
			{
				"label": "updated at",
				"name": "updated_at",
				"type": "date",
				"dateFormat": "D, d M y"
			}
		]
	} );

	$('#users').DataTable( {
		"dom": "Tfrtip",
		"ajax": "/api/users",
		"columns": [
			{
				"data": "email"
			},
			{
				"data": "first_name"
			},
			{
				"data": "last_name"
			},
			{
				"data": "created_at"
			},
			{
				"data": "updated_at"
			}
		],
		"tableTools": {
			"sRowSelect": "os",
			"aButtons": [
				{ "sExtends": "editor_create", "editor": editor },
				{ "sExtends": "editor_edit",   "editor": editor },
				{ "sExtends": "editor_remove", "editor": editor }
			]
		}
	} );
} );

}(jQuery));

**Comment by tamarica** [comment]
yes. it gives jason. i can see that when i just put in the url  :       /api/users

**Comment by Chumper** [comment]
I dont know how to integrate editor with datatables directly, so that is something i can not help you with.
Can you try without editor first and look if it is working and continue from there?

**Comment by tamarica** [comment]
yes sure. so i need to use a differnt javascript for that?

**Comment by Chumper** [comment]
Just follow the steps ind the readme: https://github.com/Chumper/Datatable/blob/master/README.md
Basically you just need to include the javascript files and the table generator will generate the necessary javascript for you.

so in the controller:
```php
public function getDatatable()
    {
        return Datatable::collection(new Collection(DB::select(DB::raw(
        "SELECT id,first_name  FROM users"))))
            ->showColumns('id', 'first_name')
            ->searchColumns('first_name')
            ->orderColumns('id','first_name')
            ->make();
    }
``` 

and the view

```php
{{ Datatable::table()
    ->addColumn('id','Name')
    ->setUrl(route('api.users'))
    ->render() }}
```

should be enough, so no custom javascript.

**Comment by tamarica** [comment]
which javascripts file i need? im not sure.
there is one who has only this script in it:
$(document).ready(function(){
    $('#myTable').DataTable();
});
and i need to include this as well? http://cdn.datatables.net/1.10.6/js/jquery.dataTables.min.js

**Comment by Chumper** [comment]
you just need to include this two files:
```html
    <script type="text/javascript" src="/assets/js/jquery.js"></script>
    <script type="text/javascript" src="/assets/js/jquery.dataTables.min.js"></script>
```
so jquery and datatable.js

**Comment by tamarica** [comment]
i included before this 2:
<script type="text/javascript" language="javascript" src="https://code.jquery.com/ui/1.11.2/jquery-ui.min.js"></script>
<script type="text/javascript" charset="utf-8" src="https://cdn.datatables.net/1.10.6/js/jquery.dataTables.min.js"></script>

is it ok too?

**Comment by DeeJaVu** [comment]
i had the same issue, i have fixed it by placing

    <script type="text/javascript" src="/assets/js/jquery.js"></script>
    <script type="text/javascript" src="/assets/js/jquery.dataTables.min.js"></script>

in my header instead of footer.. it's also mentioned in the readme
 
    Please note that the scripts must be located above the call to Datatable:

**Comment by timgws** [comment]
@tamarica please let us know how this is going. If you are still having problems, please send a gist of the entire HTML for the page.
